generator/src/main/java/com/scottlogic/deg/generator/restrictions/GranularityRestrictions.java

```diff
@@ -13,6 +14,11 @@ private GranularityRestrictions(int numericScale) {
         this.numericScale = numericScale;
     }
 
+    public static boolean isCorrectScale(Number o, double granularity) {
```

Comment:
Can we give `o` a better name please

Reply:
Can we use NumberUtils.coerceToBigDecimal here to save stringifying a number to turn it (back) into a BigDecimal